rfcs/testdriver-features.md

```diff
@@ -0,0 +1,58 @@
+
+## Details
+
+### Examples
+
+[html test](https://github.com/web-platform-tests/wpt/blob/117958ef4317d8ed16e9ea6ae63da30262f3b875/infrastructure/webdriver/bidi/subscription.html#L6)
+```javascript
+...
+<script src="/resources/testdriver.js?feature=bidi"></script>
+...
+```
+[javascript test](https://github.com/web-platform-tests/wpt/blob/117958ef4317d8ed16e9ea6ae63da30262f3b875/infrastructure/webdriver/bidi/subscription.window.js#L2)
+```javascript
+...
+// META: script=/resources/testdriver.js?feature=bidi
+...
+```
+
+### Required changes
+
+#### Configure testdriver.js based on the `?feature` flag
+
+The `testdriver.js` checks if the `?feature=bidi` query parameter was included using [document.currentScript.src](https://developer.mozilla.org/en-US/docs/Web/API/Document/currentScript) like in the [prototype](https://github.com/web-platform-tests/wpt/pull/49122/files#diff-1fe2b624679a3150e5c86f84682c5901b715dad750096a524e8cb23939e5590fR16). If `bidi` was not enabled, a [descriptive error will be thrown](https://github.com/web-platform-tests/wpt/pull/49122/files#diff-1fe2b624679a3150e5c86f84682c5901b715dad750096a524e8cb23939e5590fR22).
+
+#### Add `testrdriver_features` to TestharnessTest
+
+This RFC proposes adding `testrdriver_features` to [`TestharnessTest`](https://github.com/web-platform-tests/wpt/pull/49122/files#diff-8f280b64b0700ab9b4b343adabc7ff4d5ea4b1f45cb6c4bd7f50a19b21ebdb8cR169) so that executors can get the list of features declared by a test. The list is populated by parsing the `feature` query param from the `testdriver.js` imports used by a test. Currently, only the `bidi` feature is supported.
+
+#### (implementation-specific) Set up WebDriver BiDi in the test executor if needed
+
+Even though the `TestExecutor` needs information about whether to activate BiDi or not at `connect`, which happens before the runner receives the test to run, the configuration can be done at the point [the browser is configured](https://github.com/web-platform-tests/wpt/pull/49122/files#diff-0df24b5b583c460182e687f7dc7a6a79dd2cd3389bc4a96f48483f60fceb51f7R275). This data can be used later by the specific executor implementation to [decide which protocol to use](https://github.com/web-platform-tests/wpt/pull/49122/files#diff-0df24b5b583c460182e687f7dc7a6a79dd2cd3389bc4a96f48483f60fceb51f7R275).
+
+## Risks
```

Comment:
A risk that isn't documented here is that testdriver may be used in a subresource, in which case both the top-level resource and the subresource both need to specify the same set of features. This is hard to lint (since we don't have a total graph of subresource usage). The worst case scenario would be if the subresource allowed use of the bidi APIs without the top-level test opting in (e.g. in tests authored against implementations where the flag is a no-op in wptrunner). I think we can avoid this for now, since all testdriver commands go through the top level script, but that will hopefully change in the future (ironically only once we can use BiDi for wptrunner itself).

Reply:
Good point, added it.